src/RpcClient.ts

```diff
@@ -124,13 +124,24 @@ export class PostMessageRpcClient extends RpcClient {
     public close() {
         // Clean up old requests and disconnect. Note that until the popup get's closed by the user
         // it's possible to connect again though by calling init.
-        window.removeEventListener('message', this._receiveListener);
-        this._cleanUp();
         this._connected = false;
+        window.removeEventListener('message', this._receiveListener);
+        for (const [id, { reject }] of this._responseHandlers) {
```

Comment:
I'm not the biggest fan of using object destructuring yet, as the RPC lib should be a general purpose lib and we are not currently restricting the supported browsers for it. But I looked it up and support is generally very wide, so I guess it's safe?

Reply:
I think support is good enough.
Also I remember, that you commented once on another PR that destructuring would still be experimental. It's only `Rest in objects` that's still experimental though.